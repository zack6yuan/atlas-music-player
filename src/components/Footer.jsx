export default function Footer() {
  const year = new Date().getFullYear();
  return <div className="text-center p-8">&copy; {year} Atlas School</div>;
}
