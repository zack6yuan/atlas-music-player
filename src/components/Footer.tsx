// import sun from "../assets/sun.svg";
import moon from "../assets/moon.svg";

export default function Footer() {
  const year = new Date().getFullYear();
  return (
    <div className="">
        <div className="text-center p-8 sm:mt-50">
            <span className="colon">&copy; {year}</span>  Atlas School
        </div>
        <div className="theme-button text-center mb-10">
            <button id="theme-button">
                <img src={ moon } alt="dark-theme-icon" />
            </button>
        </div>
    </div>
  );
}
