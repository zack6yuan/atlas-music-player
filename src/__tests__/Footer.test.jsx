import { render, screen, fireEvent } from "@testing-library/react";
import { expect, test } from "vitest";
import Footer from "../components/Footer"

test("Footer renders correctly", () => {
    const { container } = render(
        <Footer />
    );
    expect(container).toMatchSnapshot();
});
test("Theme changes when button is clicked", () => {
    const { container } = render(
        <Footer />
    );
    const button = screen.getAllByAltText("theme-toggler")
    fireEvent.click(button[0])
    expect(container).toMatchSnapshot();
    fireEvent.click(button[0])
    expect(container).toMatchSnapshot();
})