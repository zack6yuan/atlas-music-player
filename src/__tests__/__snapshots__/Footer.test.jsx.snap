// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer renders correctly 1`] = `
<div>
  <div
    class="sm:-mt-60 md:-mt-92 2xl:-mt-130"
  >
    <div
      class="desc-text text-center p-8 sm:mt-50"
    >
      © 
      2025
       Atlas School
    </div>
    <div
      class="flex justify-center theme-button text-center sm:mb-15 sm:-mt-5 md:-mt-6 cursor-pointer"
    >
      <img
        alt="theme-toggler"
        id="theme-toggle"
        src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20stroke='%23000000'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'%20class='lucide%20lucide-moon-icon%20lucide-moon'%3e%3cpath%20d='M20.985%2012.486a9%209%200%201%201-9.473-9.472c.405-.022.617.46.402.803a6%206%200%200%200%208.268%208.268c.344-.215.825-.004.803.401'/%3e%3c/svg%3e"
      />
    </div>
  </div>
</div>
`;

exports[`Theme changes when button is clicked 1`] = `
<div>
  <div
    class="sm:-mt-60 md:-mt-92 2xl:-mt-130"
  >
    <div
      class="desc-text text-center p-8 sm:mt-50"
    >
      © 
      2025
       Atlas School
    </div>
    <div
      class="flex justify-center theme-button text-center sm:mb-15 sm:-mt-5 md:-mt-6 cursor-pointer"
    >
      <img
        alt="theme-toggler"
        id="theme-toggle"
        src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20stroke='%23000000'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'%20class='lucide%20lucide-moon-icon%20lucide-moon'%3e%3cpath%20d='M20.985%2012.486a9%209%200%201%201-9.473-9.472c.405-.022.617.46.402.803a6%206%200%200%200%208.268%208.268c.344-.215.825-.004.803.401'/%3e%3c/svg%3e"
      />
    </div>
  </div>
</div>
`;

exports[`Theme changes when button is clicked 2`] = `
<div>
  <div
    class="sm:-mt-60 md:-mt-92 2xl:-mt-130"
  >
    <div
      class="desc-text text-center p-8 sm:mt-50"
    >
      © 
      2025
       Atlas School
    </div>
    <div
      class="flex justify-center theme-button text-center sm:mb-15 sm:-mt-5 md:-mt-6 cursor-pointer"
    >
      <img
        alt="theme-toggler"
        id="theme-toggle"
        src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20stroke='%23000000'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'%20class='lucide%20lucide-moon-icon%20lucide-moon'%3e%3cpath%20d='M20.985%2012.486a9%209%200%201%201-9.473-9.472c.405-.022.617.46.402.803a6%206%200%200%200%208.268%208.268c.344-.215.825-.004.803.401'/%3e%3c/svg%3e"
      />
    </div>
  </div>
</div>
`;
